src/TestableHttpClient/Response/FunctionResponse.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace TestableHttpClient.Response;
+
+internal class FunctionResponse : ResponseBase
+{
+    private readonly Func<HttpRequestMessage, HttpResponseMessage>? httpResponseMessageFactory;
+    private readonly Action<HttpResponseMessageBuilder>? httpResponseMessageBuilderAction;
+
+    internal FunctionResponse(Func<HttpRequestMessage, HttpResponseMessage> httpResponseMessageFactory)
+    {
+        this.httpResponseMessageFactory = httpResponseMessageFactory ?? throw new ArgumentNullException(nameof(httpResponseMessageFactory));
+    }
+
+    internal FunctionResponse(Action<HttpResponseMessageBuilder> httpResponseMessageBuilderAction)
+    {
+        this.httpResponseMessageBuilderAction = httpResponseMessageBuilderAction ?? throw new ArgumentNullException(nameof(httpResponseMessageBuilderAction));
+    }
+
+    protected override HttpResponseMessage GetResponse(HttpRequestMessage requestMessage)
+    {
+        if (httpResponseMessageBuilderAction is not null)
```

Comment:
Instead of using this builder, consider deprecating this builder in favor of a response builder. On the other hand, we already have ConfiguredResponse, which helps with part of this interface.

Reply:
When changing the interface to get a context, the builder could just use the HttpResponseMessage from there, however, then it is still beter suited for the ConfiguredResponse.